org.eclipse.xtext.util/src/org/eclipse/xtext/util/PolymorphicDispatcher.java

```diff
@@ -353,7 +353,23 @@ protected RT handleAmbigousMethods(List<MethodDesc> result, Object... params) {
 		Collections.sort(cachedDescriptors, new Comparator<MethodDesc>() {
 			@Override
 			public int compare(MethodDesc o1, MethodDesc o2) {
-				return PolymorphicDispatcher.this.compare(o1, o2);
+				int compare = PolymorphicDispatcher.this.compare(o1, o2);
+				if (compare != 0) {
+					return compare;
+				}
+				Class<?>[] p1 = o1.getParameterTypes();
+				Class<?>[] p2 = o2.getParameterTypes();
+				int to = Math.min(p1.length, p2.length);
+
+				for (int i = 0; i < to; i++) {
+					final String n1 = p1[i].getName();
+					final String n2 = p2[i].getName();
+					compare = n1.compareTo(n2);
+					if (compare != 0) {
+						return compare;
+					}
+				}
+				return compare;
```

Comment:
:)
I see some problems with using the output of `toString()` as a basis for comparisons
* If `toString()` contains the output of `System#identityHashCode()`, then it's not deterministic
* `toString()` computes the full string, even though only the first `char`s may be needed for the comparison
* `toString()` involves string concatenation, which usually is not very fast, and which is not needed at all to make a comparison